pkg/providers/gcp/provider_postgres.go

```diff
@@ -203,6 +211,18 @@ func (p *PostgresProvider) reconcileCloudSQLInstance(ctx context.Context, pg *v1
 				msg := fmt.Sprintf("failed to modify cloudsql instance: %s", foundInstance.Name)
 				return nil, croType.StatusMessage(msg), errorUtil.Wrap(err, msg)
 			}
+
+			// Disable Maintenance - set Deny
+			logger.Infof("disable Maintenance - set Deny maintenance period 90 days (max allowed by GCP)")
+			modifiedInstance.Settings.DenyMaintenancePeriods = []*sqladmin.DenyMaintenancePeriod{{
+				EndDate:   time.Now().Add(time.Hour * 2160).Format("01-02-2006"),
+				StartDate: time.Now().Format("01-02-2006"),
```

Comment:
There is a very small chance that this results in >90 days due to two separate calls to `time.Now()`. This should be changed to set a variable to `time.Now()` and use this as the base for the start and end date.

Reply:
func main() {
    EndDate := time.Now().Add(time.Hour * 2160).Format("01-02-2006") // 90 days (max allowed by GCP)
    StartDate := time.Now().Format("01-02-2006")
    fmt.Println("StartDate: ", StartDate)
    fmt.Println("EndDate: ", EndDate)
}
$ go run test1.go 
StartDate:  02-15-2023
EndDate:  05-16-2023

Reply:
It's because of the processing delay between calls it can be wrong. If first time.Now() happens at day 1 11:59:59.9 the second time.Now() could be done on the next day. If you save it to a variable it'll be the same exact time and won't require two separate calls to the system clock

Reply:
yes, sure, 
Hope I understand it correctly now:
		timeNow := time.Now()
.....
		EndDate:   timeNow.Add(time.Hour * 2160).Format("01-02-2006"), 
		StartDate: timeNow.Format("01-02-2006"),
....
Done. Thank you